pkg/apis/contrail/v1alpha1/swiftproxy_types.go

```diff
@@ -25,10 +28,28 @@ type SwiftProxyConfiguration struct {
 	SwiftConfSecretName   string       `json:"swiftConfSecretName,omitempty"`
 	RingConfigMapName     string       `json:"ringConfigMapName,omitempty"`
 	Containers            []*Container `json:"containers,omitempty"`
+	SwiftServiceType      serviceType  `json:"swiftServiceType,omitempty"`
 	// Service name registered in Keystone, default "swift"
 	SwiftServiceName string `json:"swiftServiceName,omitempty"`
 }
 
+// GetServiceType returns Swift Proxy's service type.
+func (configuration SwiftProxyConfiguration) GetServiceType() corev1.ServiceType {
```

Comment:
here you don't have to type this switch since the values of servType will be correct, you can just cast it to corev1.ServiceType